src/Providers/Stripe/Customer/Service.php

```diff
@@ -89,29 +89,37 @@ public function addCard(Customer $customer, Card $card): Card
     /**
      * Deletes a customers card using the Stripe SDK.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return void
      */
-    public function deleteCard(Customer $customer, Card $card): void
+    public function deleteCard(string $customerUid, string $cardUid): void
     {
-        StripeCustomer::deleteSource($customer->uid, $card->uid);
+        $cardToken = new StripeToken($cardUid);
+
+        if ($cardToken->type === StripeToken::PAYMENT_METHOD) {
+            $paymentMethod = PaymentMethod::retrieve($cardUid);
+
+            $paymentMethod->detach();
+        } else {
+            StripeCustomer::deleteSource($customerUid, $cardUid);
+        }
```

Comment:
Are we not returning anything? No feedback whatsoever? Is this synchronous our asynchronous? What if it's failing?

Reply:
There will be an exception in that case

Reply:
A PHP Exception? We would need to add `@throws` in the docblock then.

Reply:
The thing is the Stripe\Error\InvalidRequest could happened everywhere we are calling Stripe API. So I thought it wasn't necessary to specify all those `@throws` on every method that is using the Stripe API. What do you think? Happy to add it everywhere if needed.